diff --git a/contracts/MiniMeBase.sol b/contracts/MiniMeBase.sol
index c518f8f..2ed2604 100644
--- a/contracts/MiniMeBase.sol
+++ b/contracts/MiniMeBase.sol
@@ -76,13 +76,13 @@ abstract contract MiniMeBase is Controlled, IERC20, IERC20Permit, EIP712, Nonces
     // `balances` is the map that tracks the balance of each address, in this
     //  contract when the balance changes the block number that the change
     //  occurred is also included in the map
-    mapping(address account => Checkpoint[] history) private balances;
+    mapping(address account => Checkpoint[] history) public balances;
 
     // `allowed` tracks any extra transfer rights as in all ERC20 tokens
-    mapping(address account => mapping(address authorized => uint256 amount) allowance) private allowed;
+    mapping(address account => mapping(address authorized => uint256 amount) allowance) public allowed;
 
     // Tracks the history of the `totalSupply` of the token
-    Checkpoint[] private totalSupplyHistory;
+    Checkpoint[] public totalSupplyHistory;
 
     // Flag that determines if the token is transferable or not.
     bool public transfersEnabled;
